README.md

```diff
@@ -4,37 +4,105 @@
+
+Two error scenarios are possible at this stage:
+- Message has been downloaded but the adaptor was unable to put it on the `Inbound MESH Queue`
+
+    Message has not been put on the queue, nor MESH has been instructed to delete it. Apart from error log entry, there is no side effect. Operation is retried on next MESH scan cycle.
+- Message has been downloaded and placed on `Inbound MESH Queue` but an error occurred while notifying MESH that the message can be deleted
+
+    Message has been put on the queue, but MESH has not been instructed to delete it.
+    Apart from error log entry, since the message has not been deleted from MESH, it will be fetched again on next MESH scan cycle. Since the message has been already put on `Inbound MESH Queue`, it will be processed again causing duplicates to appear on the `Outbound GP Queue`. Handling duplicates is described in the  [duplicates](#duplicates) section.
+
+Once the message is on the `Inbound MESH Queue` the Message Translator component picks each one up and performs following steps:
+
+1) Initial verification
+
+    This step parses the EDIFACT message and verifies the general structure of the message and can result in two types of errors:
+    - Interchange Header (UNB) is missing mandatory fields (sender, recipient, sequence number, NHSACK request flag).
+
+        Since the header can't be parsed, an NHSACK can't be produced to inform the laboratory about the error. This is a critical error and manual intervention is required. Message is rejected causing it to be placed back on the `Inbound MESH Queue`. It is picked up again for retry until the MQ max retry value is reached. After that, message is placed on the Dead Letter Queue and manual intervention is required.
+    - Interchange Trailer (UNZ) message count or Message Header (UNH) or Message Trailer (UNT) can't be parsed.
+
+        Since the Interchange Header was parsed, all the information required to produce an NHSACK is available. If Interchange Header has the NHSACK request flag set, an NHSACK is produced and sent back to the laboratory MESH mailbox.
+
+2) Translation
 
-The GP System will receive medical reports on the `Outbound GP Queue` in form of a FHIR DSTU3 Bundle.
+    At this step an EDIFACT has already been parsed and split into messages. Each message is translated into a FHIR Bundle. Any error occurring at this step, is logged and if an NHSACK is requested, appropriate is set in it. Because of the error, no FHIR Bundle is created for that single message. Other messages are not affected and can still be processed. At the end of this step, a summary NHSACK is generated containing information about all the successes and failures of EDIFACT messages processing.
```

Comment:
```suggestion
    At this step an EDIFACT has already been parsed and split into messages. Each message is translated into a FHIR Bundle. Any error occurring at this step, is logged and if an NHSACK is requested, appropriate information is set in it. Because of the error, no FHIR Bundle is created for that single message. Other messages are not affected and can still be processed. At the end of this step, a summary NHSACK is generated containing information about all the successes and failures of EDIFACT messages processing.
```